db/db_impl/db_impl.h

```diff
@@ -1620,7 +1627,21 @@ class DBImpl : public DB {
     InternalKey* manual_end = nullptr;   // how far we are compacting
     InternalKey tmp_storage;      // Used to keep track of compaction progress
     InternalKey tmp_storage1;     // Used to keep track of compaction progress
-    std::atomic<bool>* canceled;  // Compaction canceled by the user?
+
+    // NOTE: Calling EnableManualCompaction()/DisableManualCompaction() may
+    // overwrite the user-provided canceled variable in CompactRangeOptions.
+    // Typically, when CompactRange is being called in one thread (t1) with
+    // canceled = true, and DisableManualCompaction is being called in the
+    // other thread (t2), manual compaction is disabled normally, but if
+    // in t1, *canceled is suddenly changed into false, the manual compaction
+    // may potentially move on normally (NOT CANCELED) even if
+    // EnableManualCompaction() is not called due to the race condition of
+    // writing *canceled
+    std::atomic<bool>& canceled;  // Compaction canceled pointer reference
+    // When the user provides a canceled pointer in CompactRangeOptions, the
+    // above varaibe is the reference of the one pointed by user-provided
+    // `canceled`, otherwise, it is the reference of canceled_internal
+    std::atomic<bool> canceled_internal = false;
```

Comment:
```suggestion
    std::atomic<bool> canceled_internal_storage = false;
```
Just follow the same naming style with above variables.